### PR TITLE
Ipaddr2 cluster setup for 12sp5

### PR DESCRIPTION
**Pull request by mpagot**
# Changes description:

Code was using rootless cluster init and join mode by default. This mode is not supported in crmsh version delivered in 12sp5.
Change the code to add support for root mode:
1. Allow specifying user in function that generate ssh keys for the internal VMs. It allows calling it for root.
2. Change remote note syntax in cluster join command.
3. Add setting `IPADDR2_ROOTLESS` to control both.

# Related ticket:
https://jira.suse.com/browse/TEAM-9743

# Verification run:

### 15sp5
sle-15-SP5-SapCloud-Azure-Payg-x86_64-BuildLATEST_AZURE_SLE15_5_PAYG-ipaddr2_azure_test
- http://openqaworker15.qa.suse.cz/tests/300340 :green_circle:  result in using root mode everywhere

### 12sp5
- http://openqaworker15.qa.suse.cz/tests/300343 :green_circle: `crm cluster join` is now fine http://openqaworker15.qa.suse.cz/tests/300343#step/configure/191. Cluster is later not healthy, but issue is another issue (socat  missing) that I 'd like to handle in a different PR.  

### 15sp5
sle-15-SP5-SapCloud-Azure-Payg-x86_64-BuildLATEST_AZURE_SLE15_5_PAYG-ipaddr2_azure_test 
 - http://openqaworker15.qa.suse.cz/tests/300341 :green_circle: 
 
### 12sp5
sle-12-SP5-SapCloud-Azure-Payg-x86_64-BuildLATEST_AZURE_SLE12_5_PAYG-ipaddr2_azure_test 
 - http://openqaworker15.qa.suse.cz/tests/300344 :green-circle: `crm cluster join` is now fine http://openqaworker15.qa.suse.cz/tests/300343#step/configure/191. Cluster is later not healthy, but issue is another issue (socat  missing) that I 'd like to handle in a different PR.  

### 15sp5
sle-15-SP5-SapCloud-Azure-Payg-x86_64-BuildLATEST_AZURE_SLE15_5_PAYG-ipaddr2_azure_test 
 - http://openqaworker15.qa.suse.cz/tests/300234 :green_circle: 
 - http://openqaworker15.qa.suse.cz/tests/300342 :green_apple: 

### 12sp5
ROOTLESS mode is not supported in pacemaker available for this version.
- http://openqaworker15.qa.suse.cz/tests/300345 so as expected it fails in `crm cluster join` in https://openqaworker15.qa.suse.cz/tests/300345#step/configure/191

### CODE REVIEW

**Review by alvarocarvajald — APPROVED** [review]
LGTM. Just some typos in some comments. BTW, I really like all the comments in this code.

File: lib/sles4sap/ipaddr2.pm [hunk 1]
```diff
@@ -586,60 +602,130 @@ sub ipaddr2_internal_key_gen {
...
+}
 
-    # vm-01 first connection to vm-02
-    ipaddr2_ssh_internal(id => 1,
-        cmd => join(' ',
-            'ssh',
-            $user . '@' . ipaddr2_get_internal_vm_private_ip(id => 2),
-            $key_policy,
-            'whoami'),
+=head2
```
Comment (BillAnastasiadis) [inline lib/sles4sap/ipaddr2.pm:653]: shouldn't this be 

`=head2 ipaddr2_internal_key_authorize`

**Review by BillAnastasiadis — APPROVED** [review]
Aprt from the header comment, otherwise LGTM

File: lib/sles4sap/ipaddr2.pm [hunk 2]
```diff
@@ -540,10 +545,13 @@ sub ipaddr2_internal_key_accept {
 
     ipaddr2_internal_key_gen()
 
-Create, on the /tmp folder of the Worker, two ssh key set.
-One ssk key pair for each internal VM
-Then upload in each internal VM the ssh key pair using
-scp in Proxy mode
+Create, on the /tmp folder of the Worker, two ssh key set:
```
Comment (lpalovsky) [inline lib/sles4sap/ipaddr2.pm:548]: Plural:
s/set/sets/

File: lib/sles4sap/ipaddr2.pm [hunk 3]
```diff
@@ -540,10 +545,13 @@ sub ipaddr2_internal_key_accept {
...
+one ssk key pair for each internal VM.
```
Comment (lpalovsky) [inline lib/sles4sap/ipaddr2.pm:549]: s/ssk/ssh/

File: lib/sles4sap/ipaddr2.pm [hunk 4]
```diff
@@ -540,10 +545,13 @@ sub ipaddr2_internal_key_accept {
...
+Then upload in each internal VM the ssh key pair using scp
+and add pub key in the authorized_keys of the other internal VM.
+It also run the first connection to accept the host keys.
```
Comment (lpalovsky) [inline lib/sles4sap/ipaddr2.pm:552]: 3rd person has `s` at the end of the verb:
s/run/runs/

File: t/22_ipaddr2.t [hunk 5]
```diff
@@ -313,17 +313,32 @@ subtest '[ipaddr2_internal_key_accept] nc timeout' => sub {
...
     $ipaddr2->redefine(ipaddr2_bastion_pubip => sub { return 'Moriondo'; });
 
-    ipaddr2_create_cluster();
+    ipaddr2_cluster_create();
 
     note("\n  -->  " . join("\n  -->  ", @calls));
     ok((any { /.*41.*cluster init/ } @calls), 'crm cluster init on VM1');
     ok((any { /.*42.*cluster join/ } @calls), 'crm cluster join on VM2');
+    # by default it run in root mode
+    # so the join has not to  specify the user
```
Comment (lpalovsky) [inline t/22_ipaddr2.t:328]: Is the line break needed here? It's not a very long sentence

File: t/22_ipaddr2.t [hunk 6]
```diff
@@ -433,6 +478,52 @@ subtest '[ipaddr2_internal_key_gen]' => sub {
...
+
+    ok((any { /mv.*\/home\/EliaLocatelli.*/ } @calls), 'Move the key in the remote user home folder');
+};
+
+subtest '[ipaddr2_internal_key_gen] root' => sub {
+    my $ipaddr2 = Test::MockModule->new('sles4sap::ipaddr2', no_auto => 1);
+    $ipaddr2->redefine(ipaddr2_bastion_pubip => sub { return '1.2.3.4'; });
+    my @calls;
+    # return 1 means file does not exist.
+    #$ipaddr2->redefine(script_run => sub { push @calls, $_[0]; return 1; });
```
Comment (lpalovsky) [inline t/22_ipaddr2.t:516]: Commented out code. Is it intended?

**Review by lpalovsky — APPROVED** [review]
LGTM
Left few comments, but those are only cosmetic.

File: t/22_ipaddr2.t (hunk 6 above)
Comment (mpagot) [reply t/22_ipaddr2.t]: nope, not needed mock. Code  deleted

**Pull request merged**